collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java

```diff
@@ -217,13 +217,28 @@ private void updateOptionsMenu() {
         boolean isAtBeginning = formController.isCurrentQuestionFirstInForm() && !shouldShowRepeatGroupPicker();
 
         boolean isInRepeat = !isAtBeginning && screenIndex != null && formController.getEvent(screenIndex) == FormEntryController.EVENT_REPEAT;
-        optionsMenu.findItem(R.id.menu_delete_child).setVisible(isInRepeat).setEnabled(isInRepeat);
+        toggleDeleteButton(isInRepeat);
 
         boolean shouldShowPicker = shouldShowRepeatGroupPicker();
-        optionsMenu.findItem(R.id.menu_add_child).setVisible(shouldShowPicker).setEnabled(shouldShowPicker);
+        toggleAddButton(shouldShowPicker);
 
         boolean shouldShowGoUp = shouldShowPicker || !isAtBeginning;
-        optionsMenu.findItem(R.id.menu_go_up).setVisible(shouldShowGoUp).setEnabled(shouldShowGoUp);
+        toggleGoUpButton(shouldShowGoUp);
+    }
+
+    /** Override to disable this button. */
+    protected void toggleDeleteButton(boolean isEnabled) {
+        optionsMenu.findItem(R.id.menu_delete_child).setVisible(isEnabled).setEnabled(isEnabled);
```

Comment:
Also a similar question as before: why call both `setVisible` and `setEnabled`?  Wouldn't `setVisible` alone be enough?